### ISSUE TITLE
[CX-94] Filter GraphQL request by query name in Modify API Response rule

### ISSUE THREAD

**Issue by lazyvab** [requestly/requestly#643]
If the GraphQL request payload has an `operationName` defined, one can easily filter requests based on that in UI and modify the response accordingly.

<img width="423" alt="Screenshot 2023-05-25 at 7 19 05 PM" src="https://github.com/requestly/requestly/assets/6367566/4d41e3d5-e17e-401d-ac61-ee2f6a0584a3">

But if there is no `operationName` in the GraphQL request and the graphQL operation is derived from the query name, there is no direct option in UI to filter requests.

The only workaround is to use `Dynamic (JavaScript)` mode with the resource type selected as `REST`. In the JavaScript function `modifyResponse`, an argument `requestData` is received which can be used to filter request by query name.

For instance,
if GraphQL query is:
```
query JackSparrow {  
    hero  { 
        name    
    }
}
``` 
the request payload will be:
```
{     
    "query": "query JackSparrow {\n  hero {\n    name\n  \n} \n}",
}
```
And the JavaScript function will look like:
```javascript
function modifyResponse(args) {
  const {url, response, requestData, responseJSON} = args;

  if (requestData.query?.includes("query JackSparrow")) {
    // return custom response from this query
    console.log("Requestly: Modifying response", { query: requestData.query });
    return {...responseJSON, custom: true};
  }

  return response;
}
```

<sub>[CX-94](https://linear.app/requestly/issue/CX-94/filter-graphql-request-by-query-name-in-modify-api-response-rule)</sub>

**Comment by Charlezton** [comment]
This is exactly what I am currently missing in Requestly. The proposed workaround does not work because of 'unsafe-eval' as it is executing injected Javascript. Is this feature prioritised? 😃

**Comment by Charlezton** [comment]
Here is an example of the type of request payload I want to overrride:
`{ "query": "query getMovie($id: String) { title, genre, director }", "variables": { "id":"123456789" }}`

**Comment by lazyvab** [comment]
Thanks @Charlezton for helping us with an example. We have picked this task in the current sprint.
Hopefully, we will have the support by next week.

**Comment by Pritzomania** [comment]
Hi All, any update on this?

**Comment by wrongsahil** [comment]
Released in v23.10.22
- Now you can target GraphQL payloads with `query` in Response Rule using `Contains` Operator.

### Example
![image](https://github.com/requestly/requestly/assets/16779465/3e21b9fe-d83b-40a2-bed5-1e59f0bedff6)

Targeting using query in the above graphQL request

![image](https://github.com/requestly/requestly/assets/16779465/0409b4d2-f0ff-41e3-a987-25771372b9ff)
